python/ray/train/torch/train_loop_utils.py

```diff
@@ -67,7 +73,37 @@ def get_device() -> Union[torch.device, List[torch.device]]:
     from ray.air._internal import torch_utils
 
     record_extra_usage_tag(TagKey.TRAIN_TORCH_GET_DEVICE, "1")
-    return torch_utils.get_device()
+    return torch_utils.get_devices()[0]
+
+
+@PublicAPI(stability="alpha")
+def get_devices() -> List[torch.device]:
+    """Gets the correct torch device list configured for this process.
+
+    Assumes that `CUDA_VISIBLE_DEVICES` is set and is a
+    superset of the `ray.get_gpu_ids()`.
+
+    Example:
+        >>> # os.environ["CUDA_VISIBLE_DEVICES"] = "3,4"
+        >>> # ray.get_gpu_ids() == [3]
+        >>> # torch.cuda.is_available() == True
+        >>> # get_devices() == [torch.device("cuda:0")]
+
+        >>> # os.environ["CUDA_VISIBLE_DEVICES"] = "0,1,2,3,4"
+        >>> # ray.get_gpu_ids() == [4]
+        >>> # torch.cuda.is_available() == True
+        >>> # get_devices() == [torch.device("cuda:4")]
+
+        >>> # os.environ["CUDA_VISIBLE_DEVICES"] = "0,1,2,3,4,5"
+        >>> # ray.get_gpu_ids() == [4,5]
+        >>> # torch.cuda.is_available() == True
+        >>> # get_devices() == [torch.device("cuda:4"), torch.device("cuda:5")]
```

Comment:
Can we make this a codeblock instead if everything is commented?

Reply:
Since Ray Core allocate GPUs randomly, it'd be hard to allocate 2 specific devices(e.g. 4,5) to a ray actor. Make this a codeblock is probably possible with some workarounds but may hurt the readability.

Reply:
Oh I was thinking just `::testcode   :skipif: True`. Right now the code blocks look weird:

<img width="575" alt="Screenshot 2024-01-29 at 11 21 57 AM" src="https://github.com/ray-project/ray/assets/3887863/8c53a2bc-3199-43bf-ae68-303fb259025d">

Reply:
Ah I got you. I've updated the codeblock.